py3status/output.py

```diff
@@ -0,0 +1,269 @@
+import sys
+from json import dumps
+
+
+class OutputFormat:
+    """
+    A base class for formatting the output of py3status for various
+    different consumers
+    """
+
+    @classmethod
+    def instance_for(cls, output_format):
+        """
+        A factory for OutputFormat objects
+        """
+        supported_output_formats = {
+            "dzen2": Dzen2OutputFormat,
+            "i3bar": I3barOutputFormat,
+            "lemonbar": LemonbarOutputFormat,
+            "none": NoneOutputFormat,
+            "term": TermOutputFormat,
+            "tmux": TmuxOutputFormat,
+            "xmobar": XmobarOutputFormat,
+        }
+
+        if output_format in supported_output_formats:
+            return supported_output_formats[output_format]()
+        raise ValueError(
+            f"Invalid `output_format` attribute, should be one of `{'`, `'.join(supported_output_formats.keys())}`. Got `{output_format}`."
+        )
+
+    def __init__(self):
+        """
+        Constructor
+        """
+        self.separator = None
+
+    def format_separator(self, separator, color):
+        """
+        Produce a formatted and colorized separator for the output format,
+        if the output_format requires it, and None otherwise.
+        """
+        pass
+
+    def format(self, outputs):
+        """
+        Produce a line of output from a list of module output dictionaries
+        """
+        raise NotImplementedError()
+
+    def write_header(self, header):
+        """
+        Write the header to output, if supported by the output_format
+        """
+        raise NotImplementedError()
+
+    def write_line(self, output):
+        """
+        Write a line of py3status containing the given module output
+        """
+        raise NotImplementedError()
+
+
+class I3barOutputFormat(OutputFormat):
+    """
+    Format the output for consumption by i3bar
+    """
```

Comment:
Hmm. Should it be `class I3BarOutputFormat(SeparatedOutputFormat)` instead of returning `" | "` there in  added `get_default_separator()`? I did this and still can't get i3bar to print customized separator even with changed `" | "` string.

Reply:
I don't think i3bar supports a custom output separator, but it should honor the color_separator config option in the general section.

Reply:
i3status supports this option for the other supported output formats that are not i3bar

Reply:
Not an issue anymore. This is due to different default output between i3status and py3status in the terminal.